snappi_ixnetwork/snappi_api.py

```diff
@@ -284,26 +308,91 @@ def _protocols_exists(self):
         if len(topos) > 0:
             dgs = topos.DeviceGroup.find()
             if len(dgs) > 0:
-                eth_dev = len(self._ixnetwork.Topology.find()
-                              .DeviceGroup.find()
-                              .Ethernet.find())
+                eth_dev = len(
+                    self._ixnetwork.Topology.find()
+                    .DeviceGroup.find()
+                    .Ethernet.find()
+                )
                 ethv4v6_dev_count = ethv4v6_dev_count + eth_dev
                 if eth_dev > 0:
-                    v4_dev = len(self._ixnetwork.Topology.find()
-                                 .DeviceGroup.find()
-                                 .Ethernet.find().Ipv4.find())
+                    v4_dev = len(
+                        self._ixnetwork.Topology.find()
+                        .DeviceGroup.find()
+                        .Ethernet.find()
+                        .Ipv4.find()
+                    )
                     ethv4v6_dev_count = ethv4v6_dev_count + v4_dev
-                    v6_dev = len(self._ixnetwork.Topology.find()
-                                 .DeviceGroup.find()
-                                 .Ethernet.find().Ipv6.find())
+                    v6_dev = len(
+                        self._ixnetwork.Topology.find()
+                        .DeviceGroup.find()
+                        .Ethernet.find()
+                        .Ipv6.find()
+                    )
                     ethv4v6_dev_count = ethv4v6_dev_count + v6_dev
-        if (total_dev > ethv4v6_dev_count):
+        if total_dev > ethv4v6_dev_count:
             return True
         else:
             return False
 
+    def set_control_state(self, payload):
+        try:
+            control_option = payload.choice
+            control_obj = getattr(payload, control_option)
+            control_choice = control_obj.get("choice")
+            request_payload = getattr(control_obj, control_choice)
+            self._connect()
+            if control_option == "port":
```

Comment:
Add error / warning if request  for `control_option` and `control_choice` which we are not supporting.

Reply:
done.